### PR TITLE
GH-3267 OpenAPI spec for REST API

### PR DESCRIPTION
**Pull request by abrokenjester**
GitHub issue resolved: #3267  

Briefly describe the changes proposed in this PR:

- updated old swagger yaml to openapi 3.0 spec draft
- added static webpage that renders the spec as redoc on the website
- removed old hand-written rest api documentation: openapi spec is a replacement

This openapi spec can possible be improved further, both from a code-generation/testing pov as from a documentation pov, but I'd like us to switch over this ASAP as it will be easier to maintain and improve than the old hand-written docs.

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Review by barthanssens — CHANGES_REQUESTED** [review]
Not an OpenAPI expert, but it seems that the rdf4j-openapi.yaml has a few errors (i.e. copy/pasting it into https://editor.swagger.io/ produces errors on the XML-examples and some issues on properties). So perhaps something to look into...

**Comment by abrokenjester** [comment]
> Not an OpenAPI expert, but it seems that the rdf4j-openapi.yaml has a few errors (i.e. copy/pasting it into https://editor.swagger.io/ produces errors on the XML-examples and some issues on properties). So perhaps something to look into...

Argh, I thought I had fixed all those. Good catch, I'll tidy that up.

File: site/content/documentation/reference/_index.md [hunk 1]
```diff
@@ -3,5 +3,5 @@ title: "Reference"
 weight: 4
 ---
 
+- [RDF4J REST API](/documentation/reference/rest-api/)
 - [RDF4J API Javadoc](/javadoc/latest)
-- older releases: [3.7.7](/javadoc/3.7.7), [3.6.3](/javadoc/3.6.3), [3.5.0](/javadoc/3.5.0), [3.4.3](/javadoc/3.4.3), [3.3.0](/javadoc/3.3.0), [3.2.0](/javadoc/3.2.0), [3.1.0](/javadoc/3.1.0), [3.0.4](/javadoc/3.0.4), [2.5.4](/javadoc/2.5.4)
```
Comment (hmottestad) [inline site/content/documentation/reference/_index.md:7]: 3.7.7 is probably still relevant since it's the newest version for those stuck on Java 8.

**Review by hmottestad — COMMENTED** [review]
I need to run this up locally to see what it looks like. I'm glad you've taken the time to modernise our API documentation :)

File: site/content/documentation/reference/_index.md (hunk 1 above)
Comment (abrokenjester) [reply site/content/documentation/reference/_index.md:7]: Ah yeah it's not that I want to remove the actual javadocs, per se. But this long list of links is not really necessary IMO (plus, we always forget to update it anyway - not how any releases post 3.7.7 are missing).

**Comment by abrokenjester** [comment]
> I need to run this up locally to see what it looks like. I'm glad you've taken the time to modernise our API documentation :)

Tip for others who may not know this: if you have `hugo` installed on your local machine, you can just run `hugo server --disableFastRender` in the `site/` directory, and you can then view the site at http://localhost:1313/.

**Comment by abrokenjester** [comment]
I'm going to merge this this weekend unless there are major objections. I want to merge while I have time to monitor (and if necessary fix) the website (unsure if our use of the redoc script may trigger one of our site security rules, if so I'll need to tweak that to keep the site from breaking).

**Pull request merged**